x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilderTests.java

```diff
@@ -232,6 +232,12 @@ public void testToQuery() throws IOException {
 
     private void testDoToQuery(SparseVectorQueryBuilder queryBuilder, SearchExecutionContext context) throws IOException {
         Query query = queryBuilder.doToQuery(context);
+
+        // test query builder can randomly have no vectors, which rewrites to a MatchNoneQuery - nothing more to do in this case.
+        if (query instanceof MatchNoDocsQuery) {
```

Comment:
If it rewrites to `MatchNoneQuery`, does it mean the test will show as passed without any more validation? Will that be a problem in the future or `SparseVectorQueryWrapper` will never write as `MatchNoneQuery` so there won't be any risk of skipping this test?

Reply:
`SparseVectorQueryBuilder` will rewrite to a `MatchNoneQuery` if `queryVectors` is null. As you can see in the test method, this is definitely something that can happen when we create the test index: 

```
private SparseVectorQueryBuilder createTestQueryBuilder(TokenPruningConfig tokenPruningConfig) {
        SparseVectorQueryBuilder builder;
        if (randomBoolean()) {
            builder = new SparseVectorQueryBuilder(
                SPARSE_VECTOR_FIELD,
                null,
                randomAlphaOfLength(10),
                randomAlphaOfLengthBetween(10, 25),
                tokenPruningConfig != null,
                tokenPruningConfig
            );
        } else {
            builder = new SparseVectorQueryBuilder(
                SPARSE_VECTOR_FIELD,
                WEIGHTED_TOKENS,
                null,
                null,
                tokenPruningConfig != null,
                tokenPruningConfig
            );
        }
```

Reply:
Got it, thank you for the explanation